modules/xmpp/JingleSessionPC.js

```diff
@@ -695,9 +707,20 @@ export default class JingleSessionPC extends JingleSession {
      * @param success callback called when sRD/sLD cycle finishes successfully.
      * @param failure callback called with an error object as an argument if we
      *        fail at any point during setRD, createAnswer, setLD.
+     * @param {Array<JitsiLocalTrack>} [localTracks] the optional list of
+     * the local tracks that will be added, before the offer/answer cycle
+     * executes (for the local track addition to be an atomic operation together
+     * with the offer/answer).
```

Comment:
i'm thinking we add here:
```
We allow the localTracks to optionally be passed in so that the addition of the local tracks and the processing of the initial offer can all be done atomically.  The existing addLocalTracks method queues up its own modificationQueue operation, meaning we can't call it AND process the initial offer as part of the same atomic operation.
```